projects/components/src/action-menu/action-search.provider.ts

```diff
@@ -0,0 +1,120 @@
+/*!
```

Comment:
Better not to place this class under action-menu, since it is not related to it, but rather to the ActionItem only